beta/experience-platform-extension/offerdecisioning/offer-api.md

```diff
@@ -0,0 +1,180 @@
+This API retrieves the prefetched offers for the targeted decision scopes from the cache. The returned  dictionary will only contains offers for decision scopes that has already been prefetched and cached. If a certain decision scope has not been prefetched before, it won't be contained in the returned dictionary.
+
+{% tabs %}
+{% tab title="iOS" %}
+
+#### Syntax
+
+```swift
+func retrievePrefetchedOffers(decisionScopes: [DecisionScope], completionHandler: @escaping ([DecisionScope: [Offer]], Error?) -> Void)
+```
+
+* _callback_ is invoked with the `[DecisionScope: [Offer]] ` type of  value. An `Error` will be returned if SDK fails to retrieve the offers.
+
+#### Examples
+
+Here are the examples in Objective C and Swift:
+
+#### **Objective C**
+
+```objectivec
+AEPDecisionScope* homeDecisionScope = [[AEPDecisionScope alloc] initWithActivityId:@"xcore:offer-activity:124e8bc413c888dd" placementId:@"xcore:offer-placement:124e8a16430888db"];
+
+[AEPMobileOfferDecisioning retrievePrefetchedOffersWithDecisionScopes:@[homeDecisionScope] completionHandler:^(NSDictionary<AEPDecisionScope *,NSArray<AEPOffer*>*>* offers, NSError * error) {
+          // get the offers for homeDecisionScope
+  		    NSArray<AEPOffer*>* homeOffers = offers[homeDecisionScope]
+          // deal with the offers
+}];
+        
+```
+
+**Swift**
+
+```swift
+let homeDecisionScope = DecisionScope(activityId: "xcore:offer-activity:124e8bc413c888dd", placementId: "xcore:offer-placement:124e8a16430888db")
+
+OfferDecisioning.retrievePrefetchedOffers(decisionScopes: [homeDecisionScope]) { offersDict, _ in
+
+                    // get the offers for homeDecisionScope
+                    if let homeOffers = offersDict[homeDecisionScope] {
+                        // deal with the offers
+                    }
+                }
+```
+{% endtab %}
+
+
+
+{% endtabs %}
+
+## Public classes
+
+{% tabs %}
+
+
+{% tab title="iOS" %}
+### DecisionScope/**AEPDecisionScope**
+
+This class contains the id of activity and placement, wich is used by the offer decisioning service to propose offers for.
+
+```swift
+public class DecisionScope{
+    public let activityId: String
+    public let placementId: String
+  
+    public init(activityId: String, placementId: String) {
+        self.activityId = activityId
+        self.placementId = placementId
+    }
+}
+```
+
+
+
+### Offer/AEPOffer
+
+This class represents the offer option received from the offer decisioning service.
+
+```swift
+public class Offer: NSObject, Codable {
+    public let id: String
+    public let type: OfferType
+    public let language: [String]?
+    public let content: String?
+    public let characteristics: [String: String]?
+    public let schema: String
+}
+```
+
+
+
+### OfferType/AEPOfferType
+
+An enum indicates the type of an offer, which is a field of the `Offer` class
+
+```swift
+public enum OfferType{
+    case unknown = 0
+    case json = 1
+    case text = 2
+    case html = 3
+    case image = 4
+}
```

Comment:
Can we not just write 
```
public enum OfferType: Int {
    case unknown
    case json
    case text
    case html
    case image
}
```
Swift will automatically assign 0 to unknown and count upwards from there.

Reply:
this class needs to be objective-c compatible, i haven't looked much into this but followed the way how we define enums, I guess the explicit int is needed to support obj-c. I didn't include all that tech detail in the docs .

Reply:
Yes that's correct. Enum cases in swift don't have integer values set by default unlike obj-C . So once we define an enum to be of a particular type, say Int we do not need to assign a raw value to each case (Implicit assignment). Also the first case starts with 0 (unless specified) and the rest of the cases count upwards. Anyways, I'll leave it up to you.